frontend/src/lib/utils/eventUsageLogic.ts

```diff
@@ -370,18 +375,24 @@ export const eventUsageLogic = kea<
             }
 
             properties.total_event_actions_count = (properties.events_count || 0) + (properties.actions_count || 0)
+            // If we're aggregating this query by groups
+            properties.aggregating_by_groups = filters.aggregation_group_type_index != undefined
+            // If groups are being used in this query
+            properties.using_groups =
+                hasGroupProperties(filters.properties) || filters.breakdown_group_type_index != undefined
 
             let totalEventActionFilters = 0
-            filters.events?.forEach((event) => {
-                if (event.properties?.length) {
-                    totalEventActionFilters += event.properties.length
+            const entities = (filters.events || []).concat(filters.actions || [])
+            entities.forEach((entity) => {
```

Comment:
We already iterate through the event/action list on `sanitizeFilterParams`, further we have most of the insight-agnostic code there, worth moving?

Reply:
What is the intent of the sanitize fn? I was assuming it ties into the rest of the app, hence doing it here.

Reply:
This sanitize function is only used for event reporting purposes. It takes a filters object and extracts valuable insights while preserving user's privacy.